### PR TITLE
add builtwith as a source

### LINKED ISSUE #959: Add BuiltWith Domain API as a source
See: #945

### PR DESCRIPTION
**Pull request by dogancanbakir**
Closes #959

```console
$ go run . -d builtwith.com -s builtwith -v

               __    _____           __         
   _______  __/ /_  / __(_)___  ____/ /__  _____
  / ___/ / / / __ \/ /_/ / __ \/ __  / _ \/ ___/
 (__  ) /_/ / /_/ / __/ / / / / /_/ /  __/ /    
/____/\__,_/_.___/_/ /_/_/ /_/\__,_/\___/_/

                projectdiscovery.io

[INF] Current subfinder version v2.6.3-dev (development)
[INF] Loading provider config from /home/vscode/.config/subfinder/provider-config.yaml
[DBG] API key(s) found for builtwith.
[DBG] Selected source(s) for this search: builtwith
[INF] Enumerating subdomains for builtwith.com
[builtwith] .builtwith.com
[builtwith] api.builtwith.com
[builtwith] blog.builtwith.com
[builtwith] cc.builtwith.com
[builtwith] cdn-cf.builtwith.com
[builtwith] kb.builtwith.com
[builtwith] leadsdiscovery.builtwith.com
[builtwith] leadseye.builtwith.com
[builtwith] mail.builtwith.com
[builtwith] monty.builtwith.com
[builtwith] optimizer.builtwith.com
[builtwith] pm-bounces.builtwith.com
[builtwith] pro.builtwith.com
[builtwith] sendy.builtwith.com
[builtwith] trends.builtwith.com
[builtwith] trendspro.builtwith.com
trends.builtwith.com
trendspro.builtwith.com
api.builtwith.com
kb.builtwith.com
mail.builtwith.com
pm-bounces.builtwith.com
sendy.builtwith.com
monty.builtwith.com
optimizer.builtwith.com
pro.builtwith.com
.builtwith.com
cdn-cf.builtwith.com
leadsdiscovery.builtwith.com
leadseye.builtwith.com
blog.builtwith.com
cc.builtwith.com
[INF] Found 16 subdomains for builtwith.com in 1 second 283 milliseconds
```

Note: The API key works with only `-d builtwith.com` /shrug

### CODE REVIEW

File: v2/pkg/subscraping/sources/builtwith/builtwith.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+		resp, err := session.SimpleGet(ctx, fmt.Sprintf("https://api.builtwith.com/v21/api.json?KEY=%s&HIDETEXT=yes&HIDEDL=yes&NOLIVE=yes&NOMETA=yes&NOPII=yes&NOATTR=yes&LOOKUP=%s", randomApiKey, domain))
+		if err != nil {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: err}
+			s.errors++
+			session.DiscardHTTPResponse(resp)
+			return
+		}
+
+		// x, _ := io.ReadAll(resp.Body)
```
Comment (Mzack9999) [inline v2/pkg/subscraping/sources/builtwith/builtwith.go:66]: unused debug instructions

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Could you provide an example query, I was able to get an api key, but unfortunately it's unclear which is the expected format:
```console
$ go run . -d google -s builtwith -v
...
[INF] Found 0 subdomains for google.com in 459 milliseconds 533 microseconds
$ go run . -d google.com -s builtwith -v
...
[INF] Found 0 subdomains for google in 420 milliseconds 752 microseconds
...
```

I think it's also worth updating the docs

**Comment by dogancanbakir** [comment]
@Mzack9999 I have updated the initial message with an example and have also made changes to the document.

**Review by Mzack9999 — APPROVED** [review]
lgtm - Apparently `google.com` had really zero results xD

**Pull request merged**